### PR TITLE
Priest Changes

### PR DESCRIPTION
**Pull request by K1ngSpad3z**
Changed:

-Removed Healing
-Added Zeal defs
-Added Prayer balance

To Do:

-Add Zeal affs
-Look at fixing prayer balance
-create function for benediction

### CODE REVIEW

File: svo (setup, misc, empty, funnies, dor).xml [hunk 1]
```diff
@@ -728,9 +725,9 @@ if not svo.systemloaded then
   conf.ai_resetherbbal      = 2.5 -- normally at 1.6
   conf.ai_resetsalvebal     = 5
   conf.ai_resetmossbal      = 10  -- resets at 6
-  if svo.haveskillset('healing') then
-    -- resets at 2s for healing allies, near 4s for healing yourself, and offensive skills as inbetween
-    conf.ai_resethealingbal        = 7
+  if svo.haveskillset('zeal') then
+    -- resets at 3 for blessings, 4 for benediction 1.6-2.3 for affs
+    conf.ai_resetprayerbal        = 1.7
```
Comment (vadi2) [inline svo (setup, misc, empty, funnies, dor).xml]: 1.7 seems super low? It means the balance will always reset at 1.7s as a failsafe, regardless of how long it actually takes

File: svo (trigger functions).xml [hunk 2]
```diff
@@ -4663,15 +4601,15 @@ function svo.valid.pyradius()
...
-  lifevision.add(actions.stolebalance_happened.p, nil, 'healing')
+  lifevision.add(actions.stolebalance_happened.p, nil, 'zeal')
 end
 
-function svo.valid.gothealingbalance()
+function svo.valid.gotprayerbalance()
   svo.checkaction(svo.dict.gotbalance.happened, true)
-  svo.dict.gotbalance.happened.tempmap[#svo.dict.gotbalance.happened.tempmap+1] = 'healing' -- hack to allow multiple balances at once
+  svo.dict.gotbalance.happened.tempmap[#svo.dict.gotbalance.happened.tempmap+1] = 'zeal' -- hack to allow multiple balances at once
   lifevision.add(actions.gotbalance_happened.p)
```
Comment (DazedxJoker) [inline svo (trigger functions).xml]: You are creating these functions twice.

File: svo (trigger functions).xml [hunk 3]
```diff
@@ -4717,6 +4655,19 @@ function svo.valid.gotvoicebalance()
...
+  lifevision.add(actions.stolebalance_happened.p, nil, 'prayer')
+end
+
+function svo.valid.gotprayerbalance()
+  svo.checkaction(svo.dict.gotbalance.happened, true)
+  svo.dict.gotbalance.happened.tempmap[#svo.dict.gotbalance.happened.tempmap+1] = 'prayer'
+  lifevision.add(actions.gotbalance_happened.p)
+end
+end
+
```
Comment (DazedxJoker) [inline svo (trigger functions).xml]: This is the second set.

File: svo (setup, misc, empty, funnies, dor).xml [hunk 4]
```diff
@@ -1137,11 +1130,9 @@ svo.index_map = pl.tablex.index_map
 -- lostbal_focus, lostbal_herb, lostbal_salve, lostbal_purgative, lostbal_sip
 sk.salvetick, sk.herbtick, sk.focustick, sk.teatick = 0, 0, 0, 0
 sk.purgativetick, sk.siptick, sk.mosstick, sk.dragonhealtick = 0, 0, 0, 0
-sk.smoketick, sk.voicetick, sk.wordtick =  0, 0, 0
+sk.smoketick, sk.voicetick, sk.wordtick, sk.prayertick =  0, 0, 0, 0
+
```
Comment (DazedxJoker) [inline svo (setup, misc, empty, funnies, dor).xml]: You're also creating the prayertick without checking for the Zeal skillset here.

File: svo (setup, misc, empty, funnies, dor).xml (hunk 1 above)
Comment (DazedxJoker) [reply svo (setup, misc, empty, funnies, dor).xml]: This variable isn't actually referenced anywhere in the system...

Should probably add it to svo.lostbal_prayer() for the tempTimer time.

File: svo (actions dictionary).xml [hunk 5]
```diff
@@ -13693,6 +13502,13 @@ end
 if svo.haveskillset('spirituality') then
   basicdef('heresy', "hunt heresy")
 end
+if svo.haveskillset('zeal') then
+  basicdef('thermalblessing', 'recite fire me', false, nil, true)
+  basicdef('willpowerblessing', 'recite will me', false, nil, true)
+  basicdef('frostblessing', 'recite void me', false, nil, true)
+  basicdef('enduranceblessing', 'recite endurance me', false, nil, true)
+  basicdef('earthblessing', 'recite protection me', false, nil, true)
+end
```
Comment (DazedxJoker) [inline svo (actions dictionary).xml]: Okay, so this function here is overwriting the dictionary entries we created further up, and when it does it removed the prayer balance requirement. That is why it is spamming the prayer balance.

**Pull request closed without merging**